testgrid/tgapi/pkg/handlers/primary_join.go

```diff
@@ -42,9 +44,45 @@ func GetNodeJoinCommand(w http.ResponseWriter, r *http.Request) {
 		JSON(w, 500, err)
 		return
 	}
+	if secondaryJoin != "" {
```

Comment:
can you move all this to a function and write a unit test?

Reply:
I'm thinking to skip the unit test now, we might change the adjoin command function after that we can write the unit test